cannon/mipsevm/versions/version_test.go

```diff
@@ -15,3 +15,39 @@ func TestParseStateVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestIsSupported(t *testing.T) {
+	type TestCase struct {
+		name     string
+		input    int
+		expected bool
+	}
+	cases := []TestCase{}
+
+	maxSupportedValue := 0
+	for _, ver := range StateVersionTypes {
+		versionValue := int(ver)
+		if versionValue > maxSupportedValue {
+			maxSupportedValue = versionValue
+		}
+		if IsSupportedMultiThreaded64(ver) {
+			cases = append(cases, TestCase{name: "Supported version " + ver.String(), input: versionValue, expected: true})
+		} else {
+			cases = append(cases, TestCase{name: "Unsupported version " + ver.String(), input: versionValue, expected: false})
+		}
+	}
+
+	cases = append(cases,
+		TestCase{name: "Min unsupported version", input: maxSupportedValue + 1, expected: false},
+		TestCase{name: "Min unsupported version + 1", input: maxSupportedValue + 2, expected: false},
```

Comment:
Why is the name "min unsupported" and the input "maxSupported"?

Reply:
There is an interpretation of this which makes sense:

Given:
supported_versions = {1,2,3}
unsupported_versions = {4,5,6}
Then:
max-supported = 3
min-unsupported = 4 = max-supported+1